### PR TITLE
added features to Contact, minor fix to Book

### PR DESCRIPTION
**Pull request by jjyhj**
added field clearing feature to contact. Contact and Book now using string values instead of hard coded values

### CODE REVIEW

File: LawnCare/frontend/app/src/main/java/com/redflag/lawncare/book/BookNowFragment.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ public void onFailure(@NonNull Exception e) {
 
     private boolean checkForError(EditText text, boolean errorState) {
         if("".equals(text.getText().toString())){
-            text.setError(getResources().getString(R.string.err_cant_be_empty));
+            text.setError(getString(R.string.err_cant_be_empty));
```
Comment (wrm192) [inline LawnCare/frontend/app/src/main/java/com/redflag/lawncare/book/BookNowFragment.java:99]: this has already been fixed, hasnt it?

File: LawnCare/frontend/app/src/main/java/com/redflag/lawncare/book/BookNowFragment.java [hunk 2]
```diff
@@ -108,7 +108,7 @@ private void clearError (EditText text) {
 
     private boolean validatePhoneNumber(EditText phone, boolean errorState) {
         if(!android.util.Patterns.PHONE.matcher(phone.getText().toString()).matches()){
-            phone.setError(getResources().getString(R.string.err_must_be_phone));
+            phone.setError(getString(R.string.err_must_be_phone));
```
Comment (wrm192) [inline LawnCare/frontend/app/src/main/java/com/redflag/lawncare/book/BookNowFragment.java:111]: same with this one

File: LawnCare/frontend/app/src/main/java/com/redflag/lawncare/book/BookNowFragment.java (hunk 1 above)
Comment (jjyhj) [reply LawnCare/frontend/app/src/main/java/com/redflag/lawncare/book/BookNowFragment.java:99]: oh weird. my ver had the actual string in there but the string.xml had your strings in there so i just switched them in.

**Comment by wrm192** [comment]
I'm hesitant to marge this because I'm concerned it will cause history errors with the fact that errors have been corrected in a previous commit. If it were up to date with master it should be fine, but git isn't saying it a problem so I will merge anyway. But in the future, before you make a pull request, verify that your branch isn't behind master.

**Pull request merged**